### PR TITLE
Add docs for dev and test

### PR DESCRIPTION
**Pull request by amirfefer**

### CODE REVIEW

File: docs/dev.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Development
+This doc explains
```
Comment (adiabramovitch) [inline docs/dev.md:2]: It seems like the sentence was cut short, is that on purpose?

File: docs/dev.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Development
+This doc explains 
+
+
+## UX / Components
+We use [patternfly](https://www.patternfly.org/v4/) as our primary UX guidance and patternfly-react for our main components store.
+You can find further components and utils which made by `@redhat-cloud-services`
```
Comment (adiabramovitch) [inline docs/dev.md:7]: the word "which" is not necessary here :)

File: docs/dev.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      isError,
+      error,
+      data,
+  } = useQuery(MY_QUERY_KEY, fetchSomeList);
+   ```
+   There are plenty of options to choose from and control, by defining a stale query, caching time, manual triggering, pulling and more. 
+   For further reading visit the react-query [docs](https://react-query.tanstack.com/reference/useQuery)
+
+## Client global state
+While the `react-quey` replaces the need for managing the server's data state, we still need from time to time to manage a client global state.
```
Comment (adiabramovitch) [inline docs/dev.md:38]: missing r: react-query

File: docs/dev.md (hunk 1 above)
Comment (amirfefer) [reply docs/dev.md:2]: > is that on purpose?

yes, it's still draft :)

**Review by ezr-ondrej — COMMENTED** [review]
I'm bit missing some documentation on what to store in global state and what in the component state, IMHO we can store most of the data we need locally into our Wizard component :thinking:

**Comment by amirfefer** [comment]
> I'm a bit missing some documentation on what to store in the global state and what in the component state, IMHO we can store most of the data we need locally into our Wizard component 

Propagating many local states deep in the tree could end with a mess  and optimizations issues (each step/substep in the wizard is logically a page that wraps multiple components)
The wizard is expected to grow rapidly (i.e multi-cloud), I was searching for an infrastructure that is simple to consume, well optimized,  and could support the fast-growing.

**Comment by ezr-ondrej** [comment]
> I'd say, as a rule of thumb, if you need a client state in multiple steps, then it makes sense to keep it global (like the native `React.Context` approach)

Thanks, makes sense :+1:

**Review by adiabramovitch — DISMISSED** [review]

**Comment by amirfefer** [comment]
@ezr-ondrej, I updated the global state section, is it clearer now?

**Review by ezr-ondrej — APPROVED** [review]
Thanks @amirfefer !

**Pull request merged**